tidy3d/components/simulation.py

```diff
@@ -2517,6 +2517,36 @@ def boundaries_for_zero_dims(cls, val, values):
 
         return val
 
+    @pydantic.validator("monitors", always=True)
+    @skip_if_fields_missing(["size", "boundary_spec"])
+    def boundaries_for_proj_mnts(cls, val, values):
+        """Error if periodic boundaries or bloch boundaries is used with field projection monitors in 3D simulations."""
+
+        if val is None:
+            return val
+
+        sim_size = values.get("size")
+        boundaries = values.get("boundary_spec").to_list
+        axis_names = "xyz"
+
+        is_3d_simulation = sim_size.count(0.0) == 0
+        for dim, boundary in enumerate(boundaries):
+            axis = axis_names[dim]
+            num_periodic_bloch_boundaries = sum(
+                isinstance(bnd, (Periodic, BlochBoundary)) for bnd in boundary
+            )
+            if num_periodic_bloch_boundaries > 0:
+                for monitor in val:
+                    if isinstance(monitor, AbstractFieldProjectionMonitor) and is_3d_simulation:
+                        raise SetupError(
+                            f"Using 'FieldProjectionMonitor' '{monitor.name}' "
+                            f"with periodic/Bloch boundaries along the {axis} axis would lead to incorrect results. "
```

Comment:
I think no need to print the axis as using the projection monitor with any periodic/Bloch boundaries is wrong.

`Using 'FieldProjectionMonitor' '{monitor.name}'  with periodic/Bloch boundaries will lead to incorrect results.`